application/database/migrations/2020_02_03_083755_create_products_table.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateProductsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('products', function (Blueprint $table) {
+            $table->bigIncrements('id');
+            $table->string('image_path')->nullable();
+            $table->string('name');
+            $table->string('price');
+            $table->text('description')->nullable();
+            $table->unsignedBigInteger('product_category_id');
```

Comment:
これForeignKey貼れてないですね
```suggestion
            $table->unsignedBigInteger('product_category_id');
            $table->foreign('product_category_id')->references('id')->on('product_categories');
```

Reply:
migrationファイルの第二引数のクロージャが引数で持っているBlueprintクラス何ですが、
このクラスがテーブルを生成するメソッドを持っているんでしょうか？いまいち、こいつが何なのか理解できていません。

Reply:
> このクラスがテーブルを生成するメソッドを持っているんでしょうか？

まあそんなかんじ
PhpStormで`Command`押しながらBlueprint押して
定義に飛ぶから`function string`とかで検索したら出てくる

Reply:
`Schema::create`がテーブル作成のSQL発行して`Blueprint`のクロージャで発行するSQLを構築してる

Reply:
これに関しては内部構造理解しなくても
https://readouble.com/laravel/6.x/ja/migrations.html
このリファレンス読んだほうがはやい

Reply:
リファレンス再度熟読します。